src/generator/AutoRest.Go/Extensions.cs

```diff
@@ -560,7 +562,13 @@ public static string Fields(this CompositeType compositeType)
                 &&  !String.IsNullOrEmpty((compositeType as ModelTemplateModel).NextLink))
             {
                 var nextLinkField = (compositeType as ModelTemplateModel).NextLink;
-                if (!properties.Any(p => p.Name == nextLinkField))
+                foreach (Property p in properties) {
```

Comment:
This foreach removes all dots in properties (for code to work), and if it finds the already existing next link, changes its name to how NextLink was added originally (NextLink is a ModelTemplateModel property and it is used in more places, including the razor tamplates)